### PR TITLE
Android launcher icon

### PR DESCRIPTION
**Pull request by onichandame**
This is a simple implementation for bundling the launcher icon into apk through gradle.

### CODE REVIEW

File: xbuild/src/gradle/mod.rs [hunk 1]
```diff
@@ -94,6 +96,36 @@ pub fn build(env: &BuildEnv, out: &Path) -> Result<()> {
         dependencies = dependencies,
     );
 
+    if let Some(icon_path) = env.icon.as_ref() {
+        let mut scaler = xcommon::Scaler::open(icon_path)?;
+        scaler.optimize();
+        let anydpi = res.join("mipmap-anydpi-v26");
+        std::fs::create_dir_all(&anydpi)?;
+        std::fs::write(anydpi.join("ic_launcher.xml"), IC_LAUNCHER)?;
+        let dpis = vec![
```
Comment (dvc94ch) [inline xbuild/src/gradle/mod.rs:105]: doesn't have to be a vec, an array will do

File: xbuild/src/gradle/mod.rs [hunk 2]
```diff
@@ -94,6 +96,36 @@ pub fn build(env: &BuildEnv, out: &Path) -> Result<()> {
...
+            ("m", 48),
+            ("h", 72),
+            ("xh", 96),
+            ("xxh", 144),
+            ("xxh", 192),
+            ("xxxh", 256),
+        ];
+        for dpi in dpis {
```
Comment (dvc94ch) [inline xbuild/src/gradle/mod.rs:113]:
```suggestion
        for (name, size) in dpis {
```

File: xbuild/src/gradle/ic_launcher.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
+<!--
+  <background android:drawable="@mipmap/ic_launcher_background"/>
+-->
+  <foreground android:drawable="@mipmap/ic_launcher_foreground"/>
+  <monochrome android:drawable="@mipmap/ic_launcher_monochrome"/>
+</adaptive-icon>
```
Comment (dvc94ch) [inline xbuild/src/gradle/ic_launcher.xml:8]: missing trailing new line

File: xbuild/src/gradle/ic_launcher.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<adaptive-icon xmlns:android="http://schemas.android.com/apk/res/android">
+<!--
+  <background android:drawable="@mipmap/ic_launcher_background"/>
+-->
```
Comment (dvc94ch) [inline xbuild/src/gradle/ic_launcher.xml:5]: is this comment useful, or should it be removed?

File: xbuild/Cargo.toml [hunk 5]
```diff
@@ -41,7 +41,10 @@ tar = "0.4.38"
 toml = "0.5.9"
 tracing = { version = "0.1.37", default-features = false }
 tracing-log = "0.1.3"
-tracing-subscriber = { version = "0.3.16", default-features = false, features = ["env-filter", "fmt"] }
+tracing-subscriber = { version = "0.3.16", default-features = false, features = [
+    "env-filter",
+    "fmt",
+] }
```
Comment (dvc94ch) [inline xbuild/Cargo.toml:47]: unrelated change

**Review by dvc94ch — COMMENTED** [review]
thanks!

File: xbuild/src/gradle/ic_launcher.xml (hunk 4 above)
Comment (MarijnS95) [reply xbuild/src/gradle/ic_launcher.xml:5]: If it's useful the user has to be reminded to extend `for variant in ["foreground", "monochrome"]`.

Comment (onichandame) [reply xbuild/src/gradle/ic_launcher.xml]: It's a leftover from the [template](https://icon.kitchen/) I used. For most use cases I think people don't have to use an explicit background. In addition, the other platforms do not have a background for icon. So removing it may be the easiest.

File: xbuild/src/gradle/ic_launcher.xml (hunk 3 above)
Comment (onichandame) [reply xbuild/src/gradle/ic_launcher.xml]: Added the newline but github web ui does not seem to reflect this change. 
in my ide: image

**Review by dvc94ch — APPROVED** [review]

File: xbuild/src/gradle/ic_launcher.xml (hunk 3 above)
Comment (MarijnS95) [reply xbuild/src/gradle/ic_launcher.xml]: GitHub UI only reflects when it is _not_ there:

image
